### PR TITLE
Add exists method

### LINKED ISSUE #351: feat: add index exists method
The spec and tests for this new method are described here: https://github.com/algolia/algoliasearch-client-specs-internal/pull/26

### PR DESCRIPTION
**Pull request by chloelbn**
| Q                 | A
| ----------------- | ----------
| Bug fix?          | no
| New feature?      | yes
| BC breaks?        | no     
| Related Issue     | Fix #351 
| Need Doc update   | yes

### CODE REVIEW

File: spec/client_spec.rb [hunk 1]
```diff
@@ -1279,13 +1289,13 @@ def test_browse(expected, *args)
 
   context 'DNS timeout' do
     before(:each) do
-      @client = Algolia::Client.new :application_id => ENV['ALGOLIA_APPLICATION_ID'], :api_key => ENV['ALGOLIA_API_KEY'],
```
Comment (nunomaduro) [inline spec/client_spec.rb]: Updating this credentials don’t break the CI?

File: lib/algolia/index.rb [hunk 2]
```diff
@@ -1208,6 +1208,15 @@ def export_rules(hits_per_page = 100, request_options = {}, &_block)
       res
     end
 
+    def exists
+      begin
+        get_settings
+      rescue
```
[the remark this answers is not in the document]
Comment (julienbourdeau) [reply lib/algolia/index.rb]: Yes, you can specify what exception you want to catch

```ruby
begin
  raise 'This exception will be rescued!'
rescue StandardError => e
  puts "Rescued: #{e.inspect}"
end
```

[the remark this answers is not in the document]
Comment (chloelbn) [reply lib/algolia/index.rb]: done

**Comment by chloelbn** [comment]
@nunomaduro as seen together, `exists!` method is not needed

File: spec/client_spec.rb [hunk 3]
```diff
@@ -175,6 +175,16 @@ def wait_global_key_missing(key)
     @index.delete_index rescue "not fatal"
   end
 
+  it "should tell if index exists" do
+    @index.add_object!({ :name => "John Doe", :email => "john@doe.org" }, "1")
+    expect(@index.exists).to be true
+  end
+
+  it "should tell if index does not exist" do
+    index = Algolia::Index.new('test')
```
Comment (nunomaduro) [inline spec/client_spec.rb]: You may want to use a special name here, or delete the current index. The possibilities of having a `test` name on the test cluster are high.

**Pull request merged**